src/io/parser.rs

```diff
@@ -147,13 +191,52 @@ fn parse_bare_name(input: Span<'_>) -> IntermediateResult<Span<'_>> {
     )(input)
 }
 
-// TODO: parsing of DSV and Program should be unified
-/// Parse a constant from a DSV File
-#[traced("parse_dsv_constant")]
-pub fn parse_dsv_constant(input: Span<'_>) -> IntermediateResult<String> {
-    map(alt((sparql::iriref, parse_bare_name)), |iri| {
-        format!("<{iri}>")
-    })(input)
+/// Parse an IRI representing a constant.
+fn parse_iri_constant<'a>(
+    prefixes: &'a RefCell<HashMap<&'a str, &'a str>>,
+) -> impl FnMut(Span<'a>) -> IntermediateResult<'a, Identifier> {
+    map_error(
+        move |input| {
+            let (remainder, position) = position(input)?;
+            let (remainder, name) = traced(
+                "parse_iri_constant",
+                alt((
+                    map(sparql::iriref, |name| sparql::Name::IriReference(&name)),
+                    sparql::prefixed_name,
+                    sparql::blank_node_label,
+                    map(parse_bare_name, |name| sparql::Name::IriReference(&name)),
+                )),
+            )(remainder)?;
+
+            let resolved = resolve_prefixed_name(&prefixes.borrow(), name)
+                .map_err(|e| Err::Failure(e.at(position)))?;
+
+            Ok((remainder, Identifier(resolved)))
+        },
+        || ParseError::ExpectedIriConstant,
+    )
+}
+
+/// Parse a ground term.
+pub fn parse_ground_term<'a>(
+    prefixes: &'a RefCell<HashMap<&'a str, &'a str>>,
+) -> impl FnMut(Span<'a>) -> IntermediateResult<'a, Term> {
+    traced(
+        "parse_ground_term",
+        map_error(
+            alt((
+                map(parse_iri_constant(prefixes), Term::Constant),
+                map(turtle::numeric_literal, Term::NumericLiteral),
+                map(turtle::rdf_literal, move |literal| {
+                    Term::RdfLiteral(resolve_prefixed_rdf_literal(&prefixes.borrow(), literal))
+                }),
+                map(turtle::string_literal, move |literal| {
```

Comment:
Is there any reason we need an extra function here instead of just calling `turtle::string` (which is named for the actual production rule from the turtle grammar, so should also not be renamed)?

Reply:
I mindlessly copy pasted. I will change this :)